### PR TITLE
Add document "Use Cases and Gap Analysis: Frame Accuracy for Media Production"

### PR DESCRIPTION
**Pull request by gesinger**
This document collects use cases and gap analysis for frame accuracy for media production from other documents and discussions. It is meant to serve as a starting place for adding/removing/and changing such requirements as more are discovered and the scope is better defined.

Let me know if anything should be changed. And thank you.

### CODE REVIEW

File: media-production/use-cases-and-gap-analysis-frame-accurate-sync.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+This document collects use cases and requirements for new or changed Web APIs related to frame accuracy to improve support for media production.
+
+## Functional Gaps
+
+### The media timeline is not precise enough to identify individual frames
+
+[HTML] does not expose any precise mechanism to assess the time, from a user's wall clock perspective, at which a particular media frame is going to be rendered. A web application may only infer this information by looking at the [media element]'s [currentTime] property to infer the frame being rendered and the time at which the user will see the next frame. This has several limitations:
+
+* [currentTime] is represented as a double value, which does not allow it to identify individual frames due to rounding errors. This is a [known issue][html#109].
+* [currentTime] is updated at a user-agent defined rate (typically the rate at which [time marches on] runs), and is kept stable while scripts are running. When a web application reads [currentTime], it cannot tell when this property was last updated, and thus cannot reliably assess whether this property still represents the frame currently being rendered.
```
Comment (nigelmegitt) [inline media-production/use-cases-and-gap-analysis-frame-accurate-sync.md]: I'm not clear whether the request is to update `currentTime` while scripts are running or pause media progress until scripts have completed?

File: media-production/use-cases-and-gap-analysis-frame-accurate-sync.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+In addition, [currentTime] only accepts a time value, and neither has the abiliy, nor is there an alternative API, to set its value to a frame number or [SMPTE Time Code].
```
Comment (nigelmegitt) [inline media-production/use-cases-and-gap-analysis-frame-accurate-sync.md]: If the frame rate is not consistent, I don't think frame number addressing would make any sense. We should discuss if we really need SMPTE Time Code.

File: media-production/use-cases-and-gap-analysis-frame-accurate-sync.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+### Seeking to next/previous frame
+
+The [media element] does not provide a mechanism to seek by individual frames. This can be worked around by using the media's frame rate and the [media element]'s [currentTime] property to seek by a frame duration from the reported [currentTime] value, but the [currentTime] property does not guarantee frame level precision. In addition, the frame rate of the media may vary over time, may be rounded internally by the browser, and is not exposed.
```
Comment (nigelmegitt) [inline media-production/use-cases-and-gap-analysis-frame-accurate-sync.md]: If you asked "advance by 1 frame" across a frame rate change, would it be acceptable for the presentation to advance by 1 frame but the new `currentTime` to reflect the time according to the combined frame rates of all the frames including this new frame? I.e. mathematically integrate frame rate over the number of frames.

File: media-production/use-cases-and-gap-analysis-frame-accurate-sync.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+### Indeterminate frame boundaries for segments in Media Source Extensions
+
+When appending segments using [Media Source Extensions], the [timestampOffset] property does not provide enough precision to identify frame boundaries. This suffers the same limitation as [currentTime], where the value is represented as a double, which does not allow it to identify individual frames due to rounding errors.
+
+### Association of a video frame with local wall clock
+
+A [media element]'s clock is not required to follow any clock available via the [HTML] API. As such, there's no way to associate that clock with a local clock, preventing synchronization across media and other content.
```
Comment (nigelmegitt) [inline media-production/use-cases-and-gap-analysis-frame-accurate-sync.md]: This feels like a separate use case and it is not obvious to me why there is a need to do this - can we perhaps describe the outcomes that we need to achieve somewhere and then relate this list of problems to why we cannot achieve those outcomes?

File: media-production/use-cases-and-gap-analysis-frame-accurate-sync.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Use Cases and Gap Analysis: Frame Accuracy for Media Production
+
+This document collects use cases and requirements for new or changed Web APIs related to frame accuracy to improve support for media production.
+
+## Functional Gaps
+
+### The media timeline is not precise enough to identify individual frames
+
+[HTML] does not expose any precise mechanism to assess the time, from a user's wall clock perspective, at which a particular media frame is going to be rendered. A web application may only infer this information by looking at the [media element]'s [currentTime] property to infer the frame being rendered and the time at which the user will see the next frame. This has several limitations:
```
[the remark this answers is not in the document]
Comment (tidoust) [reply media-production/use-cases-and-gap-analysis-frame-accurate-sync.md]: Here and in most cases in a such document, I think this should be the [monotonic clock](https://www.w3.org/TR/hr-time/#sec-monotonic-clock) defined in High Resolutime Time; in other words, the clock seen by web applications.

File: media-production/use-cases-and-gap-analysis-frame-accurate-sync.md (hunk 1 above)
Comment (tidoust) [reply media-production/use-cases-and-gap-analysis-frame-accurate-sync.md]: A third option would be to expose the timestamp (using the `Performance.now()` clock) at which `currentTime` was last updated. `Performance.now()` keeps running while scripts are running, so web apps could run the computation themselves. In any case, I read it more as a gap than as a request for a particular solution.

**Comment by tidoust** [comment]
As discussed, the discussion has moved to a separate repository and I'm closing this issue accordingly:
https://github.com/w3c/me-media-production/

An outline of a "Requirements for Media Production" document is live at:
https://w3c.github.io/me-media-production/

It now needs to be filled out with concrete use cases, which people are encouraged to submit!

**Pull request closed without merging**